### PR TITLE
 Feature: Allow dismissal of paused Mini Player

### PR DESCRIPTION
**Pull request by pichfl**
- Dismissing the Mini Player by swiping down stops the book
- Stopping a book now removes the playing state from its cell
- A playing book can not be dismissed

Re #112

### CODE REVIEW

File: BookPlayer/Library/RootViewController.swift [hunk 1]
```diff
@@ -91,28 +105,102 @@ class RootViewController: UIViewController {
...
         guard let userInfo = notification.userInfo,
             let book = userInfo["book"] as? Book else {
                 return
         }
 
-        setupFooter(book: book)
+        setupMiniPlayer(book: book)
+    }
+
+    @objc private func bookDeleted(_ notification: Notification) {
```
Comment (GianniCarlo) [inline BookPlayer/Library/RootViewController.swift]: it seems like the notification `bookDeleted` isn't being actually used

File: BookPlayer/Library/BaseListViewController.swift [hunk 2]
```diff
@@ -81,6 +82,19 @@ class BaseListViewController: UIViewController {
         bookCell.playbackState = .paused
     }
 
+    @objc func onBookStop(_ notification: Notification) {
+        guard
+            let userInfo = notification.userInfo,
+            let book = userInfo["book"] as? Book,
+            let index = self.library.itemIndex(with: book.fileURL),
```
Comment (GianniCarlo) [inline BookPlayer/Library/BaseListViewController.swift]: This needs to be overridden in the `PlaylistViewController` to get the index inside the playlist

File: BookPlayer/Library/RootViewController.swift (hunk 1 above)
Comment (pichfl) [reply BookPlayer/Library/RootViewController.swift]: 😯 oh right, I forgot the second part of the ticket. Deleting a book should stop the book as well.

File: BookPlayer/Library/BaseListViewController.swift (hunk 2 above)
Comment (pichfl) [reply BookPlayer/Library/BaseListViewController.swift]: Done.

**Pull request merged**